### PR TITLE
Don't use pointer receivers for PubKeyMultisigThreshold

### PR DESCRIPTION
**Pull request by alessio**
Closes: #3101

* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [x] Updated CHANGELOG_PENDING.md

### CODE REVIEW

**Comment by Liamsi** [comment]
This looks good. But is there an issue for the rationale behind this change?

**Comment by Liamsi** [comment]
@alessio: I couldn't reproduce the problem we discussed in the call 🤔 Started writing a quick test and both serialization and deserialization work fine. I've added this test to `crypto/multisig/threshold_pubkey_test.go`to understand what is going on:

```go
func TestPubKeyMultisigThresholdAminoRountrip(t *testing.T) {
	msg := []byte{1, 2, 3, 4}
	pubkeys, _ := generatePubKeysAndSignatures(5, msg)
	multisigKey := NewPubKeyMultisigThreshold(2, pubkeys)
	ab, err := cdc.MarshalBinaryBare(multisigKey)
	require.NoError(t, err)

	pkmt := PubKeyMultisigThreshold{}
	err = cdc.UnmarshalBinaryBare(ab, &pkmt)
	require.NoError(t, err)

	require.Equal(t, multisigKey, &pkmt)
}
```
Maybe, I misunderstood the problem?

**Comment by Liamsi** [comment]
OK, I see. The problem is that decoding into the interface `crypto.PubKey` fails for `PubKeyMultisigThreshold `. This is happens when `cosmos-sdk/client/keys.GetKeyInfo` calls `readInfo` which decodes into a `localInfo` with a `crypto.PubKey` field. 
A failing test is in https://github.com/tendermint/tendermint/tree/multisig_decode_issue_test. The easiest would be to cherry-pick: 8eef647bfb59c7f8d340dbd66627024cc346aa50

**Comment by Liamsi** [comment]
The failing test is unrelated to this change: `TestReactorVotingPowerChange`. I can't restart the circle build (don't have the rights).

**Comment by ebuchman** [comment]
@alessio are you ok with this being on develop and waiting for next release (includes breaking changes to priv_validator.json file) or should we prepare a hot fix with this?

**Comment by jackzampolin** [comment]
@ebuchman A tag would be nice!! :pray

**Comment by alessio** [comment]
@ebuchman 1. merge to develop is bare minimum 2. release tag is better 3. give me a hotfix release and Friday drinks will be on me

File: crypto/ed25519/ed25519.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ var _ crypto.PrivKey = PrivKeyEd25519{}
 
 const (
 	PrivKeyAminoRoute = "tendermint/PrivKeyEd25519"
```
Comment (Liamsi) [inline crypto/ed25519/ed25519.go]: If we rename `*AminoRoute` -> `*AminoName`, we should rather to do consistently

File: crypto/encoding/amino/amino.go [hunk 2]
```diff
@@ -12,11 +12,11 @@ import (
 
 var cdc = amino.NewCodec()
 
-// routeTable is used to map public key concrete types back
+// nameTable is used to map public key concrete types back
 // to their amino routes. This should eventually be handled
```
Comment (Liamsi) [inline crypto/encoding/amino/amino.go]:
```suggestion
// to their registered amino names. This should eventually be handled
```

File: crypto/secp256k1/secp256k1.go [hunk 3]
```diff
@@ -17,15 +17,15 @@ import (
 //-------------------------------------
 const (
 	PrivKeyAminoRoute = "tendermint/PrivKeySecp256k1"
```
Comment (Liamsi) [inline crypto/secp256k1/secp256k1.go]: same as https://github.com/tendermint/tendermint/pull/3100/files#r246923481

**Pull request merged**